## Customer Segmentation

**Objective of This Phase**

Customer segmentation answers this business question:

**“Which groups of customers behave similarly in terms of value, usage, and risk—and how should the bank treat them differently?”**

This directly addresses the business problem:
- Banks lose revenue by treating all customers equally
- Segmentation enables targeted retention, pricing, and engagement strategies

## Business Rule–Based Segmentation

#### 1. Select Segmentation Features
They represent value, engagement, and risk.

In [2]:
import pandas as pd

df = pd.read_csv(r"C:\Users\KIIT\Desktop\Professional\Projects\BFSI Project\output\final\customer_churn_profitability_dataset.csv")


#### 2. Define High / Medium / Low Value Thresholds

In [3]:
high_value_threshold = df["total_spend"].quantile(0.75)
low_value_threshold = df["total_spend"].quantile(0.25)


#### 3. Create Value Segments

In [4]:
def value_segment(spend):
    if spend >= high_value_threshold:
        return "High Value"
    elif spend <= low_value_threshold:
        return "Low Value"
    else:
        return "Medium Value"

df["value_segment"] = df["total_spend"].apply(value_segment)


Business Meaning

- High Value → revenue drivers

- Medium Value → growth potential

- Low Value → cost-sensitive

#### 4. Create Risk Segments (Reuse Churn Logic)

In [5]:
df["risk_segment"] = df["churn_risk_level"]


#### 5. Engagement Segmentation

In [6]:
txn_median = df["transaction_count"].median()

df["engagement_segment"] = df["transaction_count"].apply(
    lambda x: "High Engagement" if x >= txn_median else "Low Engagement"
)


#### 6. Combine Segments → Final Customer Segments

In [7]:
def customer_segment(row):
    if row["value_segment"] == "High Value" and row["risk_segment"] == "High Risk":
        return "High-Value At-Risk"
    elif row["value_segment"] == "High Value" and row["risk_segment"] == "Low Risk":
        return "High-Value Loyal"
    elif row["value_segment"] == "Medium Value" and row["risk_segment"] == "High Risk":
        return "Growth At-Risk"
    elif row["value_segment"] == "Low Value" and row["risk_segment"] == "High Risk":
        return "Low-Value At-Risk"
    else:
        return "Stable / Regular"

df["customer_segment"] = df.apply(customer_segment, axis=1)


#### 7. Segment Distribution Check

In [8]:
df["customer_segment"].value_counts(normalize=True)


customer_segment
Stable / Regular      0.7185
Growth At-Risk        0.1085
High-Value Loyal      0.0745
Low-Value At-Risk     0.0505
High-Value At-Risk    0.0480
Name: proportion, dtype: float64

In [1]:
df.to_csv("customer_segmentation_output.csv", index=False)


NameError: name 'df' is not defined

## 📊 Customer Segment Distribution – Key Insights

### Segment Breakdown
- **Stable / Regular:** 71.85%
- **Growth At-Risk:** 10.85%
- **High-Value Loyal:** 7.45%
- **Low-Value At-Risk:** 5.05%
- **High-Value At-Risk:** 4.80%

---

### 1️⃣ Stable / Regular (71.85%) — *Operational Backbone*
**What this means**
- Majority of customers show stable usage, moderate value, and low churn risk.
- This distribution is typical for BFSI portfolios and indicates a healthy customer base.

**Business implication**
- Avoid heavy retention spending.
- Focus on smooth service, digital nudges, and light cross-sell opportunities.

---

### 2️⃣ Growth At-Risk (10.85%) — *Highest ROI Opportunity*
**What this means**
- Medium-value customers with early churn signals.
- Still economically viable to retain.

**Why this matters**
- Largest actionable risk group.
- Cheaper to retain than high-value at-risk customers.

**Recommended actions**
- Engagement campaigns
- Spend-based incentives
- Credit limit optimization

---

### 3️⃣ High-Value Loyal (7.45%) — *Revenue Core*
**What this means**
- High spend, low churn risk, strong repayment discipline.

**Business implication**
- Protect and reward these customers.

**Recommended actions**
- Loyalty rewards
- Fee waivers
- Premium card benefits

---

### 4️⃣ Low-Value At-Risk (5.05%) — *Cost-Sensitive Segment*
**What this means**
- Low spend combined with high churn risk.

**Business implication**
- Retention ROI is limited.

**Recommended actions**
- Low-cost digital nudges
- Self-service support
- Avoid aggressive incentives

---

### 5️⃣ High-Value At-Risk (4.80%) — 🚨 *Critical Revenue Risk*
**What this means**
- Small in size but high in revenue contribution.
- Elevated churn risk threatens disproportionate revenue loss.

**Why this segment is critical**
- Represents maximum revenue leakage risk.

**Recommended actions**
- Priority retention queue
- Personalized offers
- Proactive relationship management

---

### ✅ Segmentation Quality Check

| Check | Status |
|---|---|
| Majority stable customers | ✅ |
| High-risk segment not inflated | ✅ |
| High-value at-risk clearly identified | ✅ |
| Actionable growth-risk group present | ✅ |
